### ISSUE TITLE
Install instructions

### ISSUE THREAD

**Issue by IzzySoft** [flackbash/AudioAnchor#1]
If you wish, you can also point out that your app can be installed using the [F-Droid](https://f-droid.org/) client when adding my repo. I've just [added your app](https://apt.izzysoft.de/fdroid/index/apk/com.prangesoftwaresolutions.audioanchor) there (hope that's OK with you) – and as long as you keep on tagging releases and attaching the APK to them, those updates will automatically be available to users having it installed that way within ~24h (and they will be notified about the available update).

As it looks that your app would be perfectly meeting the inclusion criteria of the official F-Droid repo: if you want to have it listed there, let me know – and I'll gladly assist you getting it there.

Thanks for your app!

**Comment by flackbash** [comment]
Great! I would be happy to have it in the official F-Droid repo. I don't have any experience with the process though, so your help would be greatly appreciated. I guess I start by opening a new issue [here](https://gitlab.com/fdroid/rfp/issues)?

**Comment by IzzySoft** [comment]
Exactly. [Here](https://gitlab.com/fdroid/rfp/issues/915) to be more precise :rofl: 

While that RFP waits to be processed: you might wish to include screenshots in "fastlane format" with your repo (fastlane itself is not required, just the directory structure – see [here](https://f-droid.org/en/docs/All_About_Descriptions_Graphics_and_Screenshots/) for details). With the next tag after that, they'd be picked up. Of course, if you wish you can also maintain summary and description that way, and F-Droid can use that.

Oh, disclosure: I'm one of the F-Droid maintainers. But if you followed the initial link of this post, you've already seen that :smile:

Now fingers crossed: within a few hours, our bot will analyze your repo. I expect it to give an "all clear" (that's what my library scanner suggests – but then, mine doesn't scan the repo but the app) – unless I missed some blobs (JARs, AARs, `.so` and the like)…

**Comment by IzzySoft** [comment]
Hey, looks as good as expected! But for your own builds, you should secure your gradlew (see the bot's report).

**Comment by flackbash** [comment]
Cool, thanks for your help!

Concerning the screenshots: I found [this](https://github.com/fastlane/fastlane/blob/2.28.7/supply/README.md#images-and-screenshots). But the description of the directory structure there and the example structure [here](https://github.com/fastlane/fastlane/blob/2.28.7/supply/README.md#changelogs-whats-new) differ slightly. Would it be enough to add the following directory structure to my repo: *fastlane/metadata/en-US/images/phoneScreenshots/allMyScreenshotFiles.png*?
And do I really need an extra locale directory (e.g. en-US) if I only have one set of screenshots for all languages?

**Comment by IzzySoft** [comment]
If you don't want to use Fastlane itself, easiest is `metadata/android/en-US/phoneScreenshots/*.(png|jpg)`. If you plan to use fastlane (e.g. for Playstore publication), add `fastlane/` in front. Both variants are recognized by F-Droid.

I see no contradiction between the two links. The first just concentrates on images (all directories named there are located below the locale level). If you want you also can add changelogs (always nice to see what's new before updating).

And yes, you need a locale directory. `en-US` is used as fallback for all other languages if they don't exist – but the program logic is there must be a locale directory-level.

**Comment by flackbash** [comment]
I adjusted the directory structure - the screenshots can now be found [here](https://github.com/flackbash/AudioAnchor/tree/master/metadata/android/en-US/phoneScreenshots). Thanks for your explanations!

It's just the `android` directory which is not mentioned in the description.

**Comment by IzzySoft** [comment]
Exept for [here](https://github.com/fastlane/fastlane/blob/2.28.7/supply/README.md#changelogs-whats-new). Well, they don't expect you doing it manually, but to use fastlane to create that :wink:

Thanks!

**Comment by jubalh** [comment]
Being consistent in both examples still makes sense.

Cool! So does this mean @IzzySoft is pushing AudioAnchor into F-Droid? When can I get it? ;)

**Comment by IzzySoft** [comment]
@jubalh Well, as far as I can. Luckily it was already picked up by one of our integrators, so it should show up in F-Droid this week: Metadata have already been merged.

**Comment by jubalh** [comment]
@IzzySoft awesome news! Thanks for your work. Can I see this process somewhere? I have never looked into how F-Droid works, just used it. Would like to see how a new application is integrated by the example of AudioAnchor. Can you point me to a Pull Request or whatever is used?

**Comment by IzzySoft** [comment]
Sure, @jubalh. Start [Here](https://gitlab.com/fdroid/rfp/issues/915) (filing the "Request For Packaging" (RFP)) and follow up the links at the end:

* [RFP filed](https://gitlab.com/fdroid/rfp/issues/915) (here already with full metadata, which is the "icing variant"; you can also file it "end-user friendy")
* waiting for [the bot to check the source code repo](https://gitlab.com/fdroid/rfp/issues/915#note_146589341) (here: straight flash, no ERRORs found)
* an F-Droid "integrator" (a team member with build experience, so not me) picks it up and [creates a merge request](https://gitlab.com/fdroid/fdroiddata/merge_requests/4570) to integrate the metadata with the F-Droid catalog
* not unusual even if a team member opens the MR: another team member reviews
* the MR gets merged
* the build job finds the new metadata and builds the app
* manual check and signing of the APK created by the previous step
* on the next index creation, the app becomes available in the catalog.

Recommended reading with more details: My blog series on F-Droid.

* [F-Droid: The privacy-friendly alternative to Google Play Store](https://android.izzysoft.de/articles/named/fdroid-intro-1) (intro and end-user part)
* [F-Droid for advanced users and developers](https://android.izzysoft.de/articles/named/fdroid-intro-2) (includes "how to get my app listed at F-Droid?")
* [Your own F-Droid Repository with Repomaker](https://android.izzysoft.de/articles/named/fdroid-intro-3) (for apps that cannot make it to F-Droid or even into my repo)

**Comment by flackbash** [comment]
AudioAnchor is now available on F-Droid :)
I updated the install instructions and also created the launcher icons the way they are meant to be created with Android Studio so that the icon will hopefully be picked up by F-Droid with the next release.

One more question @IzzySoft : I added a short and a full description [here](https://github.com/flackbash/AudioAnchor/tree/master/metadata/android/en-US) to have F-Droid pick it up from there. But I'm not entirely sure that the directory structure, the naming and the content format are correct. Could you have a quick look at it?

**Comment by IzzySoft** [comment]
First congrats to the inclusion! To avoid confusion, I've removed it from my repo.

Fastlane structure looks OK to me – but then, I'm no dev, I can only go "by the books". To switch over to fastlane, you'd need to file a MR at fdroiddata anyway. For that to get merged, one of our "integrators" will take a look at it then. So I'd say: go ahead.

**Comment by flackbash** [comment]
Ok, I'll do that :)

**Comment by IzzySoft** [comment]
PS: Screenshots should be picked up automatically without any MR filed – usually with the next release being build. So if you're still in doubt and rather want to make "absolutely sure", that would be a way to check: tag a new release, wait until it's available at F-Droid, then check (website or client) whether the screenshots show up. If they do, go ahead for the rest.

**Comment by flackbash** [comment]
Good to know. Interestingly, the F-Droid Android app is showing the new description from this repo even though I haven't filed any MR. On the website, the short description has been updated too, but the full description is still the original one.

**Comment by IzzySoft** [comment]
"The ways of the Lord…" As I recently got to hear, nobody seems to be exactly sure when which Fastlane data is pulled. It should be the one belonging to the tag being build – but sometimes… as you noticed. As for website: that's often a few days behind (gets updated asynchronously).
